arch/x86/include/kernel_arch_func.h

```diff
@@ -66,20 +66,79 @@ _set_thread_return_value(struct k_thread *thread, unsigned int value)
 
 extern void k_cpu_atomic_idle(unsigned int imask);
 
-extern void _MsrWrite(unsigned int msr, u64_t msrData);
-extern u64_t _MsrRead(unsigned int msr);
+/**
+ * @brief Write to a model specific register (MSR)
+ *
+ * This function is used to write to an MSR.
+ *
+ * The definitions of the so-called  "Architectural MSRs" are contained
+ * in kernel_structs.h and have the format: IA32_XXX_MSR
+ *
+ * INTERNAL
+ * 1) The 'wrmsr' instruction was introduced in the Pentium processor; executing
+ *    this instruction on an earlier IA-32 processor will result in an invalid
+ *    opcode exception.
+ * 2) The 'wrmsr' uses the ECX, EDX, and EAX registers which matches the set of
+ *    volatile registers!
+ *
+ * @return N/A
+ */
+static inline void _x86_msr_write(unsigned int msr, u64_t data)
+{
+	__asm__ volatile (
+	    "movl %[msr], %%ecx\n\t"
+	    "movl %[data_lo], %%eax\n\t"
```

Comment:
Now that this is an inline, you can use a "c" constraint on the msr variable to force the compiler to put it there for you, probably saving the mov instruction (because the compiler can do it via register allocation instead).

I wouldn't be shocked if there was a similar trick available to get a 64 bit quantity into EDX:EAX too, but would have to look that up.